src/goto-programs/interpreter.cpp

```diff
@@ -500,7 +500,8 @@ exprt interpretert::get_value(
     }
     return result;
   }
-  if(use_non_det && (memory[offset].initialized>=0))
+  if(use_non_det &&
+     memory[offset].initialized!=memory_cellt::initializedt::WRITTEN_BEFORE_READ)
```

Comment:
`>=` would correspond to `WRITTEN_BEFORE_READ` or `UNKNOWN`
is this the case that you think is wrong?

Reply:
Yes, the new version is  correct. It seems to be a bug in the former.
The input synthesiser interprets the -1 (READ_BEFORE_WRITTEN) as inputs, and these are the ones that require nondet values.